Adding upgrade section for memory_ballast (#4543)
signalfx/splunk-otel-collector: README.md

[the default agent config](https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/config/collector/agent_config.yaml)
as a reference.

### From 0.96.1 to 0.97.0

> - `memory_ballast` is no longer effective. The garbage collection is now controlled by the soft memory limit set to 90%

Reviewer: ```suggestion
- `[memory_ballast](https://docs.splunk.com/observability/en/gdi/opentelemetry/components/memory-ballast-extension.html#memory-ballast-extension)` is no longer effective. The garbage collection is now controlled by the soft memory limit set to 90%
```

  of total memory (`SPLUNK_MEMORY_TOTAL_MIB` env var) by default.

  If you haven't customized the `memory_ballast`, just remove it from the configuration. 

  If you have customized it via `SPLUNK_BALLAST_SIZE_MIB` (or `extensions::memory_ballast::size_mib` config), you should

>   remove the `memory_ballast` extension and use `GOMEMLIMIT` env var to set a custom soft memory limit:

Reviewer 2: ```suggestion
  remove the `memory_ballast` extension and use the [`GOMEMLIMIT`](https://pkg.go.dev/runtime) the environment variable to set a custom soft memory limit:
```

>   remove the `memory_ballast` extension and use the [`GOMEMLIMIT`](https://pkg.go.dev/runtime) the environment variable to set a custom soft memory limit:

Reviewer: ```suggestion
  remove the `memory_ballast` extension and use the [`GOMEMLIMIT`](https://pkg.go.dev/runtime) environment variable to set a custom soft memory limit:
```
